### PR TITLE
DS-2941 PasswordAuthentication getSpecialGroups empty exception catchblock

### PR DESCRIPTION
**Pull request by DylanMeeus**
https://jira.duraspace.org/browse/DS-2941
- No longer hitting the 'catch' part as part of the logic to determine what happens when a user is not logged in. 
- Logging an error rather than an empty catch block, in case something else goes wrong.

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/authenticate/PasswordAuthentication.java [hunk 1]
```diff
@@ -136,7 +137,9 @@ public boolean isImplicit()
 		// ensures they are password users
 		try
 		{
-			if (EPersonServiceFactory.getInstance().getEPersonService().getPasswordHash(context.getCurrentUser()) != null && !EPersonServiceFactory.getInstance().getEPersonService().getPasswordHash(context.getCurrentUser()).toString().equals(""))
+            if (context.getCurrentUser() != null &&
+                    (EPersonServiceFactory.getInstance().getEPersonService().getPasswordHash(context.getCurrentUser()) != null &&
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/authenticate/PasswordAuthentication.java]: StringUtils.isNotBlank should safely treat null as blank, so this test isn't needed, saving us from reading one looooong line.

**Comment by DylanMeeus** [comment]
Yup @mwoodiupui . I have removed the unnecessary check.

**Comment by mwoodiupui** [comment]
+1 by inspection

**Comment by tdonohue** [comment]
+1 as well. Merging

**Pull request merged**